From 6938bceafbcf336476022c77608879752d06a271 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:30:04 -0400
Subject: [misc] genwqe: Enable access to VPD flash area

Message-id: <1413239416-7778-17-git-send-email-gduarte@redhat.com>
Patchwork-id: 97547
O-Subject: [RHEL7.1 PATCH BZ 1088491 16/28 v2] GenWQE: Enable access to VPD flash area
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: 5c5e0589038537848849fc827f5234a31a10f899

commit 5c5e0589038537848849fc827f5234a31a10f899
Author: Frank Haverkamp <haver@linux.vnet.ibm.com>
Date:   Thu Mar 20 15:11:02 2014 +0100

    GenWQE: Enable access to VPD flash area

    In addition to the two flash partitions we used so far, there is a 3rd
    one which is enabled for usage by this fix.

    Signed-off-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_dev.c b/drivers/misc/genwqe/card_dev.c
index 2c2c9cc..0d05ca7 100644
--- a/drivers/misc/genwqe/card_dev.c
+++ b/drivers/misc/genwqe/card_dev.c
@@ -531,7 +531,9 @@ static int do_flash_update(struct genwqe_file *cfile,
  case '1':
   cmdopts = 0x1C;
   break;  /* download/erase_first/part_1 */
- case 'v':  /* cmdopts = 0x0c (VPD) */
+ case 'v':
+  cmdopts = 0x0C;
+  break;  /* download/erase_first/vpd */
  default:
   return -EINVAL;
  }
@@ -665,6 +667,8 @@ static int do_flash_read(struct genwqe_file *cfile,
   cmdopts = 0x1A;
   break;  /* upload/part_1 */
  case 'v':
+  cmdopts = 0x0A;
+  break;  /* upload/vpd */
  default:
   return -EINVAL;
  }
-- 
1.7.1